spec/acceptance/generate-spec.ts

```diff
@@ -0,0 +1,201 @@
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import cli = require("../../lib/cli");
+import { ProjectConfig } from "../../lib/ProjectConfig";
+import { Util } from "../../lib/Util";
+import { deleteAll } from "../helpers/utils";
+
+describe("Generate command", () => {
+	let testFolder = path.parse(__filename).name;
+	let expectedTemplate: any;
+	let actualTemplate: any;
+
+	beforeEach(() => {
+		spyOn(console, "log");
+		spyOn(console, "error");
+
+		// test folder, w/ existing check:
+		while (fs.existsSync(`./output/${testFolder}`)) {
+			testFolder += 1;
+		}
+
+		fs.mkdirSync(`./output/${testFolder}`);
+		process.chdir(`./output/${testFolder}`);
+
+		// ~/.global/ homedir for global files:
+		fs.mkdirSync(`.global`);
+		spyOn(os, "homedir").and.returnValue(path.join(process.cwd(), ".global"));
+		const config = {};
+		ProjectConfig.setConfig(config as Config, true);
+	});
+
+	afterEach(() => {
+		delete require.cache[require.resolve(path.join(process.cwd(), ".global", "ignite-ui-cli.json"))];
+
+		process.chdir("../../");
+		deleteAll(`./output/${testFolder}`);
+		fs.rmdirSync(`./output/${testFolder}`);
+	});
+
+	it("Should correctly generate template with default values", async done => {
+		await cli.run(["generate", "template"]);
+
+		expect(fs.existsSync(path.join(process.cwd(), ".global"))).toBeTruthy();
+
+		const globalConfigPath = path.join(process.cwd(), ".global", "ignite-ui-cli.json");
+		expect(fs.existsSync(globalConfigPath)).toBeTruthy();
+
+		const templateFolderPath = path.join(process.cwd(), "custom-template");
+		const globalConfigActual = require(globalConfigPath);
+		const globalConfigExpected = {
+			customTemplates: [
+				"path:" + templateFolderPath
+			]
+		};
+		expect(globalConfigActual).toEqual(globalConfigExpected);
+		expect(fs.existsSync(templateFolderPath)).toBeTruthy();
+		expect(fs.existsSync(path.join(templateFolderPath, "files"))).toBeTruthy();
+
+		const templatePath = path.join(templateFolderPath, "template.json");
+		expect(fs.existsSync(templatePath)).toBeTruthy();
+		expectedTemplate = require(templatePath) as Template;
+		actualTemplate = {
+			components: [],
+			controlGroup: "",
+			dependencies: [],
+			description: "This is custom template with id: custom-template, created with IgniteUI CLI for jQuery",
```

Comment:
Umm, it's a "Ignite UI for JavaScript template created with Ignite UI CLI"?
Same for the others.